Code/GraphMol/FilterCatalog/Wrap/FilterCatalog.cpp

```diff
@@ -299,6 +299,13 @@ python::dict GetFlattenedFunctionalGroupHierarchyHelper(bool normalize) {
   }
   return dict;
 }
+
+std::vector<std::vector<boost::shared_ptr<const FilterCatalogEntry>>>
+RunFilterCatalogWrapper(const FilterCatalog &fc, const std::vector<std::string> &smiles, int numThreads) {
+  NOGIL nogil;
+  return RunFilterCatalog(fc, smiles, numThreads);
+}
```

Comment:
This scares me.  I had to add this wrapper to avoid a deadlock in the tests - I think creating new threads while holding the GIL messed it up somehow, and then when the Python logger tried to grab the GIL it wasn't able to, and locked up hard (I couldn't even `Ctrl+C` - had to `Ctrl-\`).

Adding the `NOGIL` wrapper fixed it, but my concern is that there are other functions that:
- Create threads while holding the GIL, and
- Aren't tested with invalid inputs, and therefore don't log.

Anything that meets those criteria runs a risk of hard lockup in production (when using either Python logger).

Reply:
It bothers me as well.  I'll take a look.

Reply:
I believe this is now resolved, right?

Reply:
Should be.  I pulled in Brian's fixes for Filter Catalog and Substruct Library, and I think those were the only places that had a similar dangerous pattern.